python/docs/source/getting_started/install.rst

```diff
@@ -165,16 +168,117 @@ To install PySpark from source, refer to |building_spark|_.
+
+
+Pandas API on Spark
+^^^^^^^^^^^^^^^^^^^
+
+Installable with ``pip install "pyspark[pandas_on_spark]"``.
+
+========= ================= =====================================================================
+Package   Supported version Note
+========= ================= =====================================================================
+`pandas`  >=1.4.4           Required for utilizing the Pandas API features in Spark.
+`pyarrow` >=10.0.0          Ensures efficient data handling and performance in Pandas operations.
+`numpy`   >=1.21            Facilitates complex numerical operations within Spark.
+========= ================= =====================================================================
+
+Additional libraries that enhance functionality but are not included in the installation packages:
+
+- **mlflow**: Enhances machine learning lifecycle management, including experiment tracking and model deployment.
+- **plotly, matplotlib**: Provide advanced plotting capabilities for visualization.
+
+
+ML
+^^
+
+Installable with ``pip install "pyspark[ml]"``.
+
+======= ================= =======================================================================
+Package Supported version Note
+======= ================= =======================================================================
+`numpy` >=1.21            Supports advanced data manipulation and algorithm implementation in ML.
+======= ================= =======================================================================
+
+Additional libraries that enhance functionality but are not included in the installation packages:
+
+- **scipy**: Essential for scientific computing and statistical functions in ML.
+- **scikit-learn**: Required for implementing machine learning algorithms.
+
+MLlib
+^^^^^
+
+Installable with ``pip install "pyspark[mllib]"``.
+
+======= ================= ====================================================================================================
+Package Supported version Note
+======= ================= ====================================================================================================
+`numpy` >=1.21            Essential for mathematical operations within MLlib, improves performance and accuracy of algorithms.
+======= ================= ====================================================================================================
+
+Additional libraries that enhance functionality but are not included in the installation packages:
+
+- **torch**: Utilized for machine learning model training on PySpark.
+- **torchvision**: Supports image and video processing within PySpark models.
+- **torcheval**: Facilitates model evaluation metrics in PySpark.
+- **deepspeed; sys_platform != 'darwin'**: Provides high-performance model training optimizations. Installable on non-Darwin systems.
+
+
+Spark Connect
+^^^^^^^^^^^^^
+
+Installable with ``pip install "pyspark[connect]"``.
```

Comment:
Shall we move this section to the top?
Also, what's the difference between installing `pyspark[connect]` and `pyspark-connect`? Should we highlight the difference here, or add a documentation link to the pyspark-connect package?

Reply:
Moved the section to the top.

IIRC `pyspark-connect` is not yet officially supported by pip install so I don't mentioned for now. May can update the documentation when it's officially released. @HyukjinKwon could you double check if I understand correctly?

Reply:
I already documented above. We should have the docs before releasing it out.
- `pyspark[connect]` with Jars
- `pyspark-connect` without jars.